quil-rs/src/program/mod.rs

```diff
@@ -35,7 +35,25 @@ pub mod graph;
 mod memory;
 pub mod type_check;
 
-pub type Result<O> = std::result::Result<O, ProgramError<O>>;
+#[derive(Debug, thiserror::Error, PartialEq)]
+pub enum ProgramError {
+    #[error("{0}")]
+    ParsingError(String),
```

Comment:
Would prefer not to have a string here. Can we use `ParseProgramError<Program>` instead?

Reply:
Done